src/mpid/ch4/src/ch4_coll_select.h

```diff
@@ -86,11 +86,8 @@ MPIDI_coll_algo_container_t *MPIDI_Allreduce_select(const void *sendbuf,
 #ifndef MPIDI_CH4_DIRECT_NETMOD
     if (comm->node_comm != NULL && MPIR_Comm_size(comm) == MPIR_Comm_size(comm->node_comm)) {
         /* All the ranks in comm are on the same node */
-        if (!((count * type_size) > MPIR_CVAR_MAX_POSIX_RELEASE_GATHER_ALLREDUCE_MSG_SIZE &&
-              MPIR_CVAR_ENUM_IS(BCAST_POSIX_INTRA_ALGORITHM, release_gather) &&
-              MPIR_CVAR_ENUM_IS(REDUCE_POSIX_INTRA_ALGORITHM, release_gather))) {
-            /* gamma is selected if the msg size is less than threshold or release_gather based
-             * shm_bcast and shm_reduce is not chosen for msg sizes larger than threshold */
+        if ((count * type_size) <= MPIR_CVAR_MAX_POSIX_RELEASE_GATHER_ALLREDUCE_MSG_SIZE) {
```

Comment:
I know this is not your code, but I have no clue what does "gamma" mean here...

Reply:
Yeah, the naming is gross.

Reply:
@pavanbalaji This condition was put long back when there was no collective selection framework and it was for peformance reasons. Now `if ((count * type_size) <= MPIR_CVAR_MAX_POSIX_RELEASE_GATHER_ALLREDUCE_MSG_SIZE)` condition should be removed as well.

Reply:
Also this CVAR should be removed - `MPIR_CVAR_MAX_POSIX_RELEASE_GATHER_ALLREDUCE_MSG_SIZE`

Reply:
I'll fix it.